platformio/src/main.cpp

```diff
@@ -20,64 +31,98 @@ int send_command(I2C& i2c, uint8_t address, uint16_t command) {
+    PinName rx = p14;
+    RN2483 lora(tx, rx);
+
+    // Wait for I²C sensor
+    i2c_1.frequency(20000);
+
+    // Join the network
+    bool joined = false;
+    while (!joined) {
+        led4 = 1;
+        printf("Joining TTN via OTAA...\n");
+        joined = lora.initOTAA(DEV_EUI, APP_EUI, APP_KEY, USE_ADR);
+        if (joined) {
+            printf("Joined TTN successfully!\n");
+        } else {
+            printf("Joining TTN failed\n");
+        }
+        led4 = 0;
 
-    for (auto i2c : i2cs) {
-        i2c.get().frequency(20000);
+        wait(5.0);
     }
 
+    // Main loop
     while(1) {
+        printf("------\nStart measurement...\n");
+
         led1 = 1;
         wait(0.2);
 
-        for (auto i2c : i2cs) {
-            // Start measurement with clock stretching and high repeatability
-            int error = send_command(i2c, SHT3X_I2C_ADDR, 0x2C06);
-            if (error) {
-                printf("i2c.write failed: %i\n", error);
-            }
+        int error;
+
+        // Start measurement with clock stretching and high repeatability
+        error = send_command(i2c_1, SHT3X_I2C_ADDR, 0x2C06);
+        if (error) {
+            printf("i2c.write failed: %i\n", error);
         }
-        // start conversion
+
+        // Start conversion
         ds18b20.start_measurement();
         wait(0.5);
 
-        for (auto i2c : i2cs) {
-            char data[6] = {};
-            int error = i2c.get().read(SHT3X_I2C_ADDR, data, 6);
-            if (error) {
-                printf("i2c.get().read failed: %i\n", error);
-            }
-
-            for(int i=0; i<6; ++i) {
-                printf("%02x", data[i]);
-            }
-            float tmp = calculate_temp(data[0], data[1]);
-            printf(" -> Temp = %.2f", tmp);
-
-            float humi = calculate_humi(data[3], data[4]);
-            printf(" Humi = %.2f\n", humi);
+        char data[6] = {};
+        error = i2c_1.read(SHT3X_I2C_ADDR, data, 6);
+        if (error) {
+            printf("i2c_1.read failed: %i\n", error);
+        }
+
+        for(int i=0; i<6; ++i) {
+            printf("%02x", data[i]);
         }
+        float sht_temp = calculate_temp(data[0], data[1]);
+        printf(" -> Temp = %.2f", sht_temp);
+
+        float sht_humi = calculate_humi(data[3], data[4]);
+        printf(" Humi = %.2f\n", sht_humi);
 
         bool timeout = ds18b20.wait_for_completion();
         if (timeout) {
             printf("Conversion timed out");
         }
 
-        float temperature = ds18b20.read_temperature();
-        printf("1-Wire Temp %.2f\n", temperature);
+        float ds_temp = ds18b20.read_temperature();
+        printf("1-Wire Temp %.2f\n", ds_temp);
 
         led1 = 0;
         wait(0.2);
+
+        // Measurement done, send it to TTN
+        led2 = 1;
+        uint8_t payload[12];
+        memcpy(payload, &ds_temp, 4);
+        memcpy(payload + 4, &sht_temp, 4);
+        memcpy(payload + 8, &sht_humi, 4);

```

Comment:
This sends the data in little endian format, doesn't it? Of so, shouldn't we use network byte order instead?